specs/core/0_beacon-chain.md

```diff
@@ -1790,17 +1768,19 @@ and perform the following updates:
 
 * Set `state.previous_epoch_calculation_slot = state.current_epoch_calculation_slot`
 * Set `state.previous_epoch_start_shard = state.current_epoch_start_shard`
-* Set `state.previous_epoch_randao_mix = state.current_epoch_randao_mix`
+* Set `state.previous_epoch_seed = state.current_epoch_seed`
 * Set `state.current_epoch_calculation_slot = state.slot`
 * Set `state.current_epoch_start_shard = (state.current_epoch_start_shard + get_current_epoch_committee_count_per_slot(state) * EPOCH_LENGTH) % SHARD_COUNT`
-* Set `state.current_epoch_randao_mix = get_randao_mix(state, state.current_epoch_calculation_slot - SEED_LOOKAHEAD)`
+* Set `state.current_epoch_seed = hash(get_randao_mix(state, state.current_epoch_calculation_slot - SEED_LOOKAHEAD) + get_active_index_root(state, state.current_epoch_calculation_slot))`
```

Comment:
I believe the invariant described below is broken here. 

Only in "Final Updates" are we updating the `active_index_root` for the the current `state.slot`. At this point in the execution, we haven't yet set `active_index_root` for `state.slot` but `state.current_epoch_calculation_slot == state.slot` and we are trying to access it. If we want the root of the current active val set, then we need to update `latest_index_roots` _before_ we attempt to access the value.

This should be safe to do at the top of the per-epoch change before the vset change. The vset change can't affect our active vset for this slot, only for `slot + ENTRY_EXIT_DELAY`